process/interceptors/multiDataInterceptor.go

```diff
@@ -130,8 +132,15 @@ func (mdi *MultiDataInterceptor) ProcessReceivedMessage(message p2p.MessageP2P,
 			continue
 		}
 
-		isForCurrentShard := interceptedData.IsForCurrentShard()
 		isWhiteListed := mdi.whiteListRequest.IsWhiteListed(interceptedData)
+		if !isWhiteListed && errOriginator != nil {
+			lastErrEncountered = errOriginator
+			allWhiteListed = false
+			wgProcess.Done()
+			continue
```

Comment:
:+1: this will filter out requested data from a received chunk that might contain unwanted data. Clever.